apps/backend/src/modules/section-statuses/section-status.repository.ts

```diff
@@ -0,0 +1,128 @@
+import { DatabaseTableName } from "~/libs/modules/database/database.js";
+import { type Repository } from "~/libs/types/types.js";
+
+import { RELATION_NAME } from "./libs/enums/enums.js";
+import { type SectionStatusGetAllRequestDto } from "./libs/types/types.js";
+import { SectionStatusEntity } from "./section-status.entity.js";
+import { type SectionStatusModel } from "./section-status.model.js";
+
+class SectionStatusRepository implements Repository<SectionStatusEntity> {
+	private sectionStatusModel: typeof SectionStatusModel;
+
+	public constructor(sectionStatusModel: typeof SectionStatusModel) {
+		this.sectionStatusModel = sectionStatusModel;
+	}
+
+	public async create(
+		courseSection: SectionStatusEntity,
+	): Promise<SectionStatusEntity> {
+		const sectionStatusModel = await this.sectionStatusModel
+			.query()
+			.insert(courseSection.toNewObject())
+			.returning("*")
+			.execute();
+
+		return SectionStatusEntity.initialize({
+			courseSectionId: sectionStatusModel.courseSectionId,
+			createdAt: sectionStatusModel.createdAt,
+			id: sectionStatusModel.id,
+			status: sectionStatusModel.status,
+			updatedAt: sectionStatusModel.updatedAt,
+			userId: sectionStatusModel.userId,
+		});
+	}
+
+	public async delete(id: number): Promise<boolean> {
+		const deletedItemsCount = await this.sectionStatusModel
+			.query()
+			.deleteById(id)
+			.execute();
+
+		return Boolean(deletedItemsCount);
+	}
+
+	public async find(id: number): Promise<SectionStatusEntity | null> {
+		const sectionStatusModel = await this.sectionStatusModel
+			.query()
+			.findById(id)
+			.execute();
+
+		if (!sectionStatusModel) {
+			return null;
+		}
+
+		return SectionStatusEntity.initialize({
+			courseSectionId: sectionStatusModel.courseSectionId,
+			createdAt: sectionStatusModel.createdAt,
+			id: sectionStatusModel.id,
+			status: sectionStatusModel.status,
+			updatedAt: sectionStatusModel.updatedAt,
+			userId: sectionStatusModel.userId,
+		});
```

Comment:
The same, but it's up to you to choose
```suggestion
      return sectionStatusModel
          ? SectionStatusEntity.initialize({
                  courseSectionId: sectionStatusModel.courseSectionId,
                  createdAt: sectionStatusModel.createdAt,
                  id: sectionStatusModel.id,
                  status: sectionStatusModel.status,
                  updatedAt: sectionStatusModel.updatedAt,
                  userId: sectionStatusModel.userId,
            })
      : null;
```

Reply:
can be both